### PR TITLE
API provides a list of articles

### PR DESCRIPTION
**Pull request by piavW**
[PT-link](https://www.pivotaltracker.com/story/show/169263150)

## PR Content
API can provide a list of articles.

## This PR Contains
- Articles model & controller
- Request test 
- Unit test for articles model, validation and factory

## What we've learned
- Deeper knowledge of namespace and using JSON.

### CODE REVIEW

**Review by AlexSaxena — APPROVED** [review]

**Review by clalil — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]

**Review by miyeaier — APPROVED** [review]

**Review by tochman — CHANGES_REQUESTED** [review]
Not ready for merge

**Comment by piavW** [comment]
> Not ready for merge

Merge conflict resolved.

File: app/controllers/v1/article_controller.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class V1::ArticleController < ApplicationController
```
Comment (clalil) [inline app/controllers/v1/article_controller.rb]: Why do you have singular for controller, and two of them? 🤔

File: app/controllers/v1/articles_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class V1::ArticlesController < ApplicationController
```
Comment (clalil) [inline app/controllers/v1/articles_controller.rb:1]: I like this controller the best. :)

File: spec/requests/v1/ApiCanDisplayListOfArticles_spec.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe 'Get ./requests/v1 ' do
```
Comment (clalil) [inline spec/requests/v1/ApiCanDisplayListOfArticles_spec.rb]: Duplicate files, please remove this one.

File: spec/requests/v1/api_can_display_list_of_articles_spec.rb [hunk 4]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe 'GET articles index' do 
+  let!(:articles) {2.times { create(:article)}}
+
```
Comment (clalil) [inline spec/requests/v1/api_can_display_list_of_articles_spec.rb]: This one seems appropriate to keep around.

File: spec/requests/v1/ApiCanDisplayListOfArticles_spec.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+require 'rails_helper'
+
+RSpec.describe 'Get ./requests/v1 ' do 
+  let!(:articles) {2.times { FactoryBot.create(:article)}}
+  
+  it 'lists a collection of articles' do 
+  get '/v1/article'
+
```
Comment (clalil) [inline spec/requests/v1/ApiCanDisplayListOfArticles_spec.rb]: Add a sad path for example if no title or content is entered, should it still allow it?

File: spec/requests/v1/api_can_display_list_of_articles_spec.rb [hunk 6]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
```
Comment (tochman) [inline spec/requests/v1/api_can_display_list_of_articles_spec.rb]: I don’t think you need this require

File: spec/requests/v1/api_can_display_list_of_articles_spec.rb [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+RSpec.describe 'GET articles index' do 
+  let!(:articles) {2.times { create(:article)}}
```
Comment (tochman) [inline spec/requests/v1/api_can_display_list_of_articles_spec.rb]: Format your code here

**Review by tochman — APPROVED** [review]
No blockers. But address my comments

**Comment by tochman** [comment]
Wait. The git history is still off. Wtf?

File: app/controllers/v1/article_controller.rb (hunk 1 above)
Comment (piavW) [reply app/controllers/v1/article_controller.rb]: removed duplicates

File: spec/requests/v1/ApiCanDisplayListOfArticles_spec.rb (hunk 5 above)
Comment (piavW) [reply spec/requests/v1/ApiCanDisplayListOfArticles_spec.rb]: We don't have a create action yet but we did one for when there are not articles.

**Review by miyeaier — APPROVED** [review]

**Review by clalil — APPROVED** [review]

**Review by jonas-masharqa — APPROVED** [review]

**Review by oliverochman — APPROVED** [review]

**Pull request merged**